assets/src/edit-story/elements/utils/useBrightnessChecker.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import ColorThief from 'colorthief';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useState, useCallback, useLayoutEffect } from '@wordpress/element';
+
+function useBrightnessChecker( limit ) {
```

Comment:
I usually prefer to use side-effectful hooks as effect-like hooks: free of state and with a handler argument. Otherwise, it can't react to the `ref` changes. Something like this is also not easily applicable in SSR which also points to it being a side-effect. WDYT?

Reply:
I have literally no idea what you're talking about. 😝

Can you dumb it down for me with an example?

Reply:
My point was simply to do this:

```
useCheckBrightnessEffect(ref, limit, callback) {
  useEffect( () => {
     // do the measurement
     callback( );
  }, [ ... ] );
}
```

And then:
```
function VisibleImage() {
  const [isTooBright, setIsTooBright] = useState( false );
  const ref = useRef();
  useCheckBrightnessEffect( ref, limit, setIsTooBright );
  return <Image ref={ref} ...>;
}
```

Not a huge difference, but IMHO the effect becomes more reusable and predictable - i.e. it's clearly a side-effect with no state.

Reply:
I can see that. I've actually went a bit further and made it a generic `useAverageColor` hook in f237607. Thanks!